src/base/server_type_enum.hpp

```diff
@@ -25,6 +25,13 @@ typedef enum
   SERVER_TYPE_TRANSACTION,
   SERVER_TYPE_PAGE,
   SERVER_TYPE_ANY,
+  SERVER_TYPE_SINGLE_NODE,
 } SERVER_TYPE;
 
+typedef enum
+{
+  TRANSACTION,
+  PAGE,
+  SINGLE_NODE,
+} SERVER_TYPE_CONFIG;
```

Comment:
Use enum class to force scoping values and avoid naming conflicts. Values will be accessed as `server_type_config::TRANSACTION` instead of simply `TRANSACTION`.
```suggestion
enum class
{
  TRANSACTION,
  PAGE,
  SINGLE_NODE,
} server_type_config;
```